service/chain/disk.go

```diff
@@ -68,6 +68,9 @@ func (d *Disk) Commit(height uint64) (flow.StateCommitment, error) {
 	if err != nil {
 		return flow.StateCommitment{}, fmt.Errorf("could not look up commit: %w", err)
 	}
+	if len(commit) != 32 {
+		return flow.StateCommitment{}, fmt.Errorf("invalid commit length: got %d want 32", len(commit))
+	}
```

Comment:
I think we can drop even this check, and only check the disk feeder. Everything in the protocol state has to be trusted, otherwise, nothing holds up at all. So the security stands and falls with the protocol state.